app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -385,8 +385,9 @@ public void onWindowVideoAvailabilityChanged(@NonNull WindowWidget aWindow) {
         // Show the launch dialogs, if needed.
         if (!showPrivacyDialogIfNeeded()) {
             showWhatsNewDialogIfNeeded();
-            mWindows.restoreSessions();
         }
+
+        mWindows.restoreSessions();
```

Comment:
OK this seems to clarify a bit a previous comment I made about having the same code twice. However I would like to understand exactly when this code and the `attachToWindow()` is triggered, could you clarify that?

Reply:
This call to `restoreSessions()` will happen as part of the activity's `onCreate` step.

The calls to `attachToWindow()` happen in the same step, and also in general whenever a new window is created. Some UI classes don't implement this method so we have to update their `parentHandle` directly.

For example, for the policy dialog we can do:

    privacyPolicyDialog.attachToWindow(mWindows.getFocusedWindow());

But for What's New we have to do:

    mWhatsNewWidget.getPlacement().parentHandle = mWindows.getFocusedWindow().getHandle();

(It would be a good idea to fix this eventually).